options/ansi/generic/stdio-stubs.cpp

```diff
@@ -1257,8 +1257,41 @@ size_t fwrite_unlocked(const void *buffer, size_t size, size_t count, FILE *file
 	}
 }
 
-char *fgets_unlocked(char *, int, FILE *) {
-	__ensure(!"Not implemented");
-	__builtin_unreachable();
-}
+char *fgets_unlocked(char *__restrict buffer, int max_size, FILE *stream) {
+    __ensure(max_size > 0);
+    for(int i = 0; ; i++) {
+        if (i == max_size - 1) {
+            buffer[i] = 0;
+            return buffer;
+        }
+
+        auto c = fgetc_unlocked(stream);
+
+        // If fgetc() fails, there is either an EOF or an I/O error.
+        if(c == EOF) {
+            // if(ferror(stream)) {
+            //  // Technically, we do not have to terminate the buffer in this case;
+            //  // do it anyway to avoid UB if apps do not check our result.
+            //  buffer[i] = 0;
+            //  return nullptr;
+            // }
+
+            // EOF is only an error if no chars have been read yet.
+            //__ensure(feof(stream));
```

Comment:
Why is this commented out?

Reply:
I don't know, I copied it from `fgets()` where it is also commented out.